### PR TITLE
[PHPUnit 10] Use `TestCase` suffix for abstract tests in `/Tests/`

### PR DESCRIPTION
**Pull request by OskarStark**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Refs #49233
| License       | MIT
| Doc PR        | -

Using `Test` suffix is deprecated since PHPUnit 10

Spotted in
* https://github.com/symfony/symfony/pull/49233

cc @derrabus

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Tests/Form/EventListener/MergeDoctrineCollectionListenerTestCase.php [hunk 1]
```diff
@@ -20,7 +20,7 @@
 use Symfony\Component\Form\FormEvents;
 use Symfony\Component\Form\FormFactoryInterface;
 
-class MergeDoctrineCollectionListenerTest extends TestCase
+class MergeDoctrineCollectionListenerTestCase extends TestCase
```
Comment (xabbuh) [inline src/Symfony/Bridge/Doctrine/Tests/Form/EventListener/MergeDoctrineCollectionListenerTestCase.php]: Is this change necessary as this class is not abstract (same for some other classes below)?

**Comment by OskarStark** [comment]
Should be good now @xabbuh

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/workflow_not_valid.php [hunk 2]
```diff
@@ -1,13 +1,13 @@
 <?php
 
-use Symfony\Bundle\FrameworkBundle\Tests\DependencyInjection\FrameworkExtensionTest;
+use Symfony\Bundle\FrameworkBundle\Tests\DependencyInjection\FrameworkExtensionTestCase;
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/Fixtures/php/workflow_not_valid.php:3]: the class name is used in the XML and YAML fixtures too

**Comment by xabbuh** [comment]
Search the codebase with `abstract class .+Test extends` I also find some more classes that probably need to be updated as well.

**Comment by derrabus** [comment]
Since we're changing the names of a lot of classes anyway, shall we make the naming a bit more consitent? We have `AbstractSomethingTestCase`, `BaseSomethingTestCase`, `SomethingTestCase`, …

I think, `SomethingTestCase` is the pattern that we use for our reusable test case classes already (`WebTestCase`, `KernelTestCase`, …). Would it make sense to use that pattern everywhere?

**Comment by nicolas-grekas** [comment]
What about doing this on 6.3. We can continue using phpunit 9 for the time being on 5.4.

**Comment by OskarStark** [comment]
@xabbuh but aren't these classes in /Test/ which follows BC?

@derrabus open for the renaming of others agree

@nicolas-grekas we are doing the other PHPUnit stuff against 5.4, but how you like

**Comment by fabpot** [comment]
I don't have a strong opinion, but I think we shouldn't do it on 5.4.

**Comment by OskarStark** [comment]
Let's go with 6.3 then 👍🏻🤓

**Comment by wouterj** [comment]
Agreed, this is a feature according to our [policies](https://symfony.com/doc/current/contributing/code/maintenance.html):

> The following changes are never accepted in a patch version:
> * ...
> * Support for newer major versions of Composer dependencies

**Comment by OskarStark** [comment]
Per se it's just a renaming of an internal class, so the policy does not match here from my POV.

The one why I thought it makes sense to use 5.4 is to avoid merge conflicts.

But as I said, ok let's go with 6.3

But what about the static data providers and the open PR from @xabbuh ?

**Comment by OskarStark** [comment]
Lets wait for
* https://github.com/rectorphp/rector-phpunit/pull/141

**Comment by OskarStark** [comment]
Superseded by
* #49253

**Comment by derrabus** [comment]
> What about doing this on 6.3. We can continue using phpunit 9 for the time being on 5.4.

Late to the party here. 🙈 

We can go for 6.3 for this change, but that means we need to silence the deprecation regarding those classes on 5.4. We should run the tests on the 5.4 branch with PHPUnit 9.6 because 9.5 likely won't receive fixes for PHP 8.3/8.4.

**Comment by nicolas-grekas** [comment]
@derrabus makes sense. I've cherry-picked #49253 on 5.4 in aec227139667fb6677d7cd6a8dd747d2ca03d321

**Comment by xabbuh** [comment]
@OskarStark see #49419
